packages/host/src/Instance.ts

```diff
@@ -583,11 +583,23 @@ rcon.print(game.table_to_json(players))`.replace(/\r?\n/g, " ");
 	 * entries from the given settings object.
 	 *
 	 * @param overrides - Server settings to override.
+	 * @param includeCredentials - Include Factorio username and token from host/controller config.
 	 * @returns
 	 *     server example settings with the given settings applied over it.
 	 */
-	async resolveServerSettings(overrides: Record<string, unknown>) {
+	async resolveServerSettings(overrides: Record<string, unknown>, includeCredentials: boolean) {
 		let serverSettings = await this.server.exampleSettings();
+		if (includeCredentials && !overrides.username && !overrides.token) {
+			let credentials = {
+				username: this._host.config.get("host.factorio_username") ?? undefined,
+				token: this._host.config.get("host.factorio_token") ?? undefined,
+			};
+			if (!credentials.username && !credentials.token) {
+				Object.assign(credentials, await this.sendTo("controller", new lib.GetFactorioCredentialsRequest()));
+			}
+			if (credentials.username) { serverSettings.username = credentials.username; }
+			if (credentials.token) { serverSettings.token = credentials.token; }
```

Comment:
When starting an instance it is not clear which credentials are being used. This makes it easy to misconfigure and hard to diagnose. This can be fixed by adding a log message when the credentials are injected into the settings file: `Using Factorio credetials from controller/host/instance config.`

Reply:
Good idea. I implemented this. And also spent way to long trying to figure out why it didn't work. Turns out it would load external credentials if factorio.settings.token is an empty string, and then overwrite the loaded credential with the empty string 🤦.

Reply:
Nice that we already caught a bug by adding logging :D